azure/scope/managedmachinepool.go

```diff
@@ -224,6 +224,52 @@ func buildAgentPoolSpec(managedControlPlane *infrav1.AzureManagedControlPlane,
 		}
 	}
 
+	if managedMachinePool.Spec.LinuxOSConfig != nil {
+		agentPoolSpec.LinuxOSConfig = &agentpools.LinuxOSConfig{
+			SwapFileSizeMB:             managedMachinePool.Spec.LinuxOSConfig.SwapFileSizeMB,
+			TransparentHugePageDefrag:  (*string)(managedMachinePool.Spec.LinuxOSConfig.TransparentHugePageDefrag),
+			TransparentHugePageEnabled: (*string)(managedMachinePool.Spec.LinuxOSConfig.TransparentHugePageEnabled),
```

Comment:
consider using the `pointers` library here and everywhere that needs pointer to/deref

```suggestion
			TransparentHugePageEnabled: pointers.String(managedMachinePool.Spec.LinuxOSConfig.TransparentHugePageEnabled),
```

Reply:
I am not sure if I can use `pointers.String()` here since I was typecasting it from `*TransparentHugePageOption` to `*string` type?

Reply:
whoops pls ignore